src/sound-demo/sound-demo.tsx

```diff
@@ -0,0 +1,289 @@
+    this.loopDuration = duration;
+    if (this.playing && this.loop) {
+      this.loop.interval = `${this.loopDuration}i`;
+    }
+  }
+
+  setNoteDuration(duration: number) {
+    this.noteDuration = duration;
+  }
+
+  updateSynth() {
+    if (this.sound instanceof Tone.Synth || this.sound instanceof Tone.PolySynth) {
+      this.sound.set({ envelope: {
+        release: this.synthReleaseTime,
+        releaseCurve: this.synthReleaseCurve
+      } });
+    }
+  }
+
+  setSynthReleaseTime(time: number) {
+    this.synthReleaseTime = time;
+    this.updateSynth();
+  }
+
+
+  setNote(note: string) {
+    this.noteText = note;
+    const frequency = Tone.Frequency(note).toFrequency();
+    if (isNaN(frequency)) {
+      console.warn("Invalid note:", note);
+      return;
+    }
+    this.note = note;
+    if (this.sound instanceof Tone.Oscillator) {
+      this.sound.frequency.value = frequency;
+    }
+  }
+
+  /**
+   * This probably only works when PolySynth is first created.
+   * So the user needs to stop playback, change max polyphony, then start playback again.
+   * @param count
+   */
+  setMaxPolyphony(count: number) {
+    this.maxPolyphony = count;
+  }
+
+  setSynthReleaseCurve(curve: string) {
+    this.synthReleaseCurve = curve as SynthCurve;
+    this.updateSynth();
+  }
+
+  async togglePlayPause() {
+    console.log("play/pause begin");
+
+    await Tone.start();
+
+    if (!this.playing) {
+      // Update our sounds in case parameters changed while stopped
+      this.updateSound();
+      const { sound } = this;
+      if (!sound) {
+        console.warn("No sound available");
+        return;
+      }
+      sound.toDestination();
+
+      this.loop = new Tone.Loop((time) => {
+        if (sound instanceof Tone.Synth || sound instanceof Tone.PolySynth) {
+          // As of Tone.js version 15, the poly synth release behavior just looks for a voice with
+          // the given note that hasn't been released yet, and then releases it. So if we are playing
+          // multiple overlapping notes that are the same frequency, tone.js might decide to release
+          // one that hasn't sounded for its full duration.
+          sound.triggerAttackRelease(this.note, `${this.noteDuration}i`, time);
+        } else {
+          sound.start(time);
+          sound.stop(time + Tone.Time(`${this.noteDuration}i`).toSeconds());
+        }
+      }, `${this.loopDuration}i`).start(0);
+
+      const transport = Tone.getTransport();
+      transport.start();
+      this.playing = true;
+    } else {
+      Tone.getTransport().stop();
+      this.playing = false;
+      this.loop?.dispose();
+    }
+    console.log("play/pause end");
+  }
+}
+
+export const SoundDemo = observer(function SoundDemo() {
+  const [looper] = useState(() => new Looper());
+
+  const handlePlayPause = useCallback(() => {
+    looper.togglePlayPause();
+  }, [looper]);
+
+  return (
+    <div>
+      <button onClick={handlePlayPause}>Play/Pause</button><br />
```

Comment:
The `<br />` here is technically not needed since the following `<div>` is a block-level element. It doesn't break anything, though, if you want to keep it for some reason.